common/src/main/java/com/sequenceiq/cloudbreak/eventbus/Promise.java

```diff
@@ -0,0 +1,65 @@
+package com.sequenceiq.cloudbreak.eventbus;
+
+import java.util.concurrent.CompletableFuture;
+import java.util.concurrent.ExecutionException;
+import java.util.concurrent.TimeUnit;
+import java.util.concurrent.TimeoutException;
+
+import com.fasterxml.jackson.annotation.JsonIgnore;
+
+public class Promise<T> {
+
+    @JsonIgnore
+    private final CompletableFuture<T> completableFuture = new CompletableFuture<>();
+
+    public static <T> Promise<T> prepare() {
+        return new Promise<>();
+    }
+
+    public void accept(T value) {
+        if (!isComplete()) {
+            completableFuture.complete(value);
+        }
+    }
+
+    public T await(long timeout, TimeUnit timeUnit) throws InterruptedException {
+        try {
+            return completableFuture.get(timeout, timeUnit);
+        } catch (TimeoutException e) {
+            throw new RuntimeException(e);
```

Comment:
we usually use some custom exceptions instead of the builtins. would it make sense to use some exact exceptions instead of pure runtimeex?

Reply:
I changed it to throw service specific exception.